### ISSUE TITLE
Bug in signature for fmpz_poly

### ISSUE THREAD

**Issue by thofma** [wbhart/flint2#230]
```
julia> using Nemo

julia> ZZx, x = ZZ["x"];

julia> f = x^13 + 1*x^7 + 1*x^5 + 1*x^3 + 1*x^2 + 1*x + 1
x^13+x^7+x^5+x^3+x^2+x+1

julia> signature(f)
(-3,8)
```

According to Magma the signature should be (1, 6).

**Comment by wbhart** [comment]
Yeah, (-3, 8) doesn't look much like a signature.

**Comment by thofma** [comment]
Fixed by #235
